packages/node-cli/src/commands/ManageAccountCommandGenerator.js

```diff
@@ -0,0 +1,263 @@
+/*
+ ** Copyright (c) 2020 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+"use strict";
+
+const BaseCommandGenerator = require("./BaseCommandGenerator");
+const SdkExecutionContext = require("../SdkExecutionContext");
+const { executeWithSpinner } = require("../ui/CliSpinner");
+const SdkOperationResultUtils = require("../utils/SdkOperationResultUtils");
+const CommandUtils = require("../utils/CommandUtils");
+const NodeTranslationService = require("../services/NodeTranslationService");
+const ManageAccountOutputFormatter = require("./outputFormatters/ManageAccountOutputFormatter");
+const AuthenticationService = require("../core/authentication/AuthenticationService");
+const { ManageAccountActionResult } = require("./actionresult/ManageAccountActionResult");
+const assert = require("assert");
+
+const inquirer = require("inquirer");
+
+const {
+   COMMAND_MANAGE_ACCOUNT: { ERRORS, QUESTIONS, QUESTIONS_CHOICES, MESSAGES },
+   YES,
+   NO,
+} = require("../services/TranslationKeys");
+
+const ACTION = {
+   EXIT: "exit",
+   RENAME: "rename",
+   REMOVE: "remove",
+   REVOKE: "revoke",
+};
+
+const COMMAND = {
+   OPTIONS: {
+      INFO: "info",
+      LIST: "list",
+      REMOVE: "remove",
+      RENAME: "rename",
+      RENAMETO: "renameto",
+   },
+};
+
+const ANSWERS_NAMES = {
+   SELECTED_AUTH_ID: "selected_auth_id",
+   ACTION: "action",
+   AUTHID: "authId",
+   RENAMETO: "renameto",
+   REMOVE: "remove",
+};
+
+const {
+   showValidationResults,
+   validateAuthIDNotInList,
+   validateAlphanumericHyphenUnderscore,
+   validateFieldHasNoSpaces,
+   validateFieldIsNotEmpty,
+   validateMaximumLength,
+   validateSameAuthID,
+} = require("../validation/InteractiveAnswersValidator");
+
+const PROPERTIES = {
+   INFO: "info",
+   ACCOUNT_INFO: "accountInfo",
+   URLS: "urls",
+};
+
+const DOMAIN = "domain";
+module.exports = class ManageAccountCommandGenerator extends BaseCommandGenerator {
+   constructor(options) {
+      super(options);
+      this._authenticationService = new AuthenticationService(options.executionPath);
+      this._outputFormatter = new ManageAccountOutputFormatter(options.consoleLogger);
+   }
+
+   async _getCommandQuestions(prompt) {
+      const authIDList = await this._authenticationService.getAuthIds(this._sdkExecutor);
+      let answers = await this._selectAuthID(authIDList, prompt);
+      this._outputFormatter.logAccountCredentials(answers[ANSWERS_NAMES.SELECTED_AUTH_ID]);
```

Comment:
I'm not sure about this line. I thought that the outputFormatter was only taking care of the final output of the command, but maybe I'm wrong. From my point of view this is like a middle step used to collect the answers.

Reply:
Javier and I were discussing just the same and concluded to leave it like this because:

For suitecloud account:manageauth -info this method would be final output... so that way we reuse code.
It's still an output, because it's showing the info of the selected account. Even if it is not final output, I give you that point.
Alternative it's to create a service only for this case and I think it's making it more complex.

However, if you still have some doubt about this we could discuss rest of the team and decide if output is only final or also middle steps.

Reply:
Changed and moved logic to new Calss AccountCredentialsService